File: 8_kyu/The_Feast_of_Many_Beasts.go
package kata

func Feast(beast string, dish string) bool {
	if len(beast) == 0 || len(dish) == 0 {
		// Error exit
		return false
	}
	beastRunes := []rune(beast)
	dishRunes := []rune(dish)
	return beastRunes[0] == dishRunes[0] && beastRunes[len(beastRunes)-1] == dishRunes[len(dishRunes)-1]
}

File: 8_kyu/The_Feast_of_Many_Beasts.c
#include <stdbool.h>
#include <stdlib.h>
#include <string.h>

// Gets last char of a string or 0 if string is empty
char lastChar( const char* str ) {
  // NULL pointer error
  if( str == NULL )
    return 0;
  // empty string
  if( *str == 0 )
    return 0;
  // goto string end marker
  while( *str != 0 ) str++;
  // last char is one char before end marker
  return str[-1];
}

// Returns true if and only if the two strings have same start and end char.
bool feast(const char beast[], const char dish[]) {
  // NULL pointer error exit
  if( beast == NULL || dish == NULL )
    return false;
  // first and last char must be the same
  return *beast == *dish && lastChar(beast) == lastChar(dish);
}
